### PR TITLE
Adds data race in docs

### PR DESCRIPTION
**Pull request by allonsy**
Thanks for all your hard work!
This is in reference to #32733 
I know there has been a discussion about this on PR #32538 so you are welcome to keep the code as is or merge my documentation in.
Let me know what you think and/or if you want me to modify anything!

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Hi, thanks for the PR! Could squash your commits please?

File: src/doc/book/concurrency.md [hunk 1]
```diff
@@ -174,7 +174,7 @@ fn main() {
 
     for i in 0..3 {
```
Comment (Manishearth) [inline src/doc/book/concurrency.md]: If i isn't being used we should try `for _ in 1..3`

Comment (GuillaumeGomez) [inline src/doc/book/concurrency.md]: `i` is used (it's set to data[0]).

Comment (Manishearth) [inline src/doc/book/concurrency.md]: ah, missed that

**Comment by allonsy** [comment]
@GuillaumeGomez 
Squashed!

**Pull request merged**